src/algebra/big_operators/basic.lean

```diff
@@ -761,6 +761,29 @@ lemma prod_range_succ' (f : ℕ → β) :
 | 0       := prod_range_succ _ _
 | (n + 1) := by rw [prod_range_succ _ n, mul_right_comm, ← prod_range_succ', prod_range_succ]
 
+lemma eventually_constant_prod {u : ℕ → β} {N : ℕ} (hu : ∀ n ≥ N, u n = 1) {n : ℕ} (hn : n ≥ N) :
+  ∏ k in range (n + 1), u k = ∏ k in range (N + 1), u k :=
+begin
+  obtain ⟨m, rfl : n = N + m⟩ := le_iff_exists_add.mp hn,
+  clear hn,
+  induction m with m hm,
+  { simp },
+  erw [prod_range_succ, hm],
+  simp [hu]
+end
+
+lemma eventually_constant_sum {β} [add_comm_monoid β] {u : ℕ → β} {N : ℕ}
+  (hu : ∀ n ≥ N, u n = 0) {n : ℕ} (hn : n ≥ N) :
+  ∑ k in range (n + 1), u k = ∑ k in range (N + 1), u k :=
+begin
+  obtain ⟨m, rfl : n = N + m⟩ := le_iff_exists_add.mp hn,
+  clear hn,
+  induction m with m hm,
+  { simp },
+  erw [sum_range_succ, hm],
+  simp [hu]
+end
```

Comment:
```suggestion
lemma eventually_constant_sum {β} [add_comm_monoid β] {u : ℕ → β} {N : ℕ}
  (hu : ∀ n ≥ N, u n = 0) {n : ℕ} (hn : n ≥ N) :
  ∑ k in range (n + 1), u k = ∑ k in range (N + 1), u k :=
@eventually_constant_prod (multiplicative β) _ _ _ hu _ hn

attribute [to_additive] eventually_constant_prod
```

Reply:
@eric-wieser from the top post:

> eventually_constant_prod and eventually_constant_sum are duplicated by hand because to_additive gets confused by the appearance of 1.

Reply:
Did you test this code? (I didn't)

Reply:
Yes, I did. It's confused if you ask it to translate the lemma, which is why I add `to_additive` manually instead.